src/transforms/groupby.js

```diff
@@ -35,6 +35,35 @@ exports.attributes = {
             'with `x` [1, 3] and one trace with `x` [2, 4].'
         ].join(' ')
     },
+    nameformat: {
```

Comment:
Yeah, this is cool! I hadn't thought about extending this to other keys but I can imagine that going off in all sorts of directions in the future (`%{marker.size*5}`???).

`%{...}` seems sufficiently useless on its own that we can do this without calling it a breaking change.